azurerm/data_source_arm_client_config_test.go

```diff
@@ -43,6 +46,22 @@ func testAzureRMClientConfigAttr(name, key, value string) resource.TestCheckFunc
 	}
 }
 
+func testAzureRMClientConfigGUIDAttr(name, key string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		r, err := regexp.Compile("^[A-Fa-f0-9]{8}-[A-Fa-f0-9]{4}-[A-Fa-f0-9]{4}-[A-Fa-f0-9]{4}-[A-Fa-f0-9]{12}$")
```

Comment:
would it make more sense to use [this method in the `go.uuid` library](http://godoc.org/github.com/satori/go.uuid#FromString) (which is already vendored) for this?